include/modules/sv_detection_methods/analyze_cigar_method.hpp

```diff
@@ -2,6 +2,96 @@
+ * \param[in]       pos_ref             - position of the inserted part in the ref (current position)
+ * \param[in]       pos_read            - position of the inserted part in the read (current position)
+ * \param[in]       inserted_bases      - the inserted bases of the possible duplication
+ * \param[in]       min_length          - minimum length of variants to detect (default 30 bp,
+ *                                                                              expected to be non-negative)
+ * \param[in, out]  pos_start_dup_seq   - start position of the duplicated seq (excluding itself)
+ * \param[in, out]  pos_end_dup_seq     - end position of the duplicated seq (including itself)
+ * \param[in, out]  tandem_dup_count    - the number of tandem copies of the inserted sequence
+ * \returns         duplicated_bases    - the duplicated bases of the duplication
+ *
+ * \details If the inserted bases include one or more copies of a duplicated sequence, which is suffix or prefix of
+ *          another copy, we have a Tandem Duplication. To check this, we have to do a semi-global alignment.
+ *
+ *          Case 1: The duplication (insertion) comes after the matched sequence. Thus we need to check if the inserted
+ *                  sequence matches (partly) the suffix sequence. The length of the matching part yields to the amount
+ *                  of copies, thus we can calculate the tandem_dup_count and the inserted_bases.
+ *
+ *                  ref AAAACCGCGTAGCGGG----------TACGTAACGGTACG
+ *                        ||||||||||||||          |||||||| -> inserted sequence: GCGGGGCGGG
+ *                  read  AACCGCGTAGCGGGGCGGGGCGGGTACGTAAC
+ *
+ *                  suffix_sequence AAAACCGCGTAGCGGG       -> free_end_gaps_sequence1_leading{true},
+ *                                             |||||          free_end_gaps_sequence1_trailing{false}
+ *                  inserted_bases             GCGGGGCGGG  -> free_end_gaps_sequence2_leading{false},
+ *                                                            free_end_gaps_sequence2_trailing{true}
+ *                  -> tandem_dup_count = 3, duplicated_bases = GCGGG
+ *
+ *          Case 2: The duplication (insertion) comes before the matched sequence.
+ *                  ref AAAACCGCGTA----------GCGGGTACGTAACGGTACG
+ *                        |||||||||          |||||||||||||  -> inserted sequence: GCGGGGCGGG
+ *                  read  AACCGCGTAGCGGGGCGGGGCGGGTACGTAAC
+ *
+ *                  prefix_sequence     GCGGGTACGTAACGGTACG -> free_end_gaps_sequence1_leading{false},
+ *                                      |||||                  free_end_gaps_sequence1_trailing{true}
+ *                  inserted_bases GCGGGGCGGG               -> free_end_gaps_sequence2_leading{true},
+ *                                                             free_end_gaps_sequence2_trailing{false}
+ *                  -> tandem_dup_count = 3, duplicated_bases = GCGGG
```

Comment:
Other Idea:
create suffix tree of the inserted sequence and search for longest common repeated substring without overlap (with errors) and than map this repeated substring (without errors?).

Other input: Burrows Wheeler, occurence table, FM index; reg Expression -> build minimal automat; ZIP Hoffmann code